### ISSUE TITLE
Backbone not frozen when evaluating linear probing ability in downstream tasks

### ISSUE THREAD

**Issue by sigureling** [BINE022/EEGPT#18]
Hello, in the EEGPT paper, it is claimed that the metrics in downstream tasks are accessed by linear probing. But in the downstream part of code, I didn't see where the backbone parameters are frozen. I also search the key word: requires_grad,but didn't find anything either. The only thing I found is that the encoder was set to eval mode. But that won't stop the gradient computing. Maybe I was careless. It will be nice if you can tell me where you freeze the backbone. Thank you。

**Comment by ejejeieei** [comment]
Yes, I also found that the backbone is not frozen.

**Comment by leoeooe** [comment]
I think the optimizer doesn't have the parameters from backbone, then the backbone is not updated in the finetune. Why the author did't set the requires_grad=false, I think it's because the backbone is between the chan_conv and linear_probe. These two modules need grad from backbone to update:

        optimizer = torch.optim.AdamW(
            list(self.chan_conv.parameters())+
            list(self.linear_probe1.parameters())+
            list(self.linear_probe2.parameters()),
            weight_decay=0.01)#

**Comment by sigureling** [comment]
Oh, I understand. I didn't notice the parameters passed into optimizer. Thanks for you kind explanation
